esp-hal/src/uart.rs

```diff
@@ -754,6 +754,26 @@ where
         }
     }
 
+    /// Sends a break signal for a specified duration in bit time, i.e. the time
+    /// it takes to transfer one bit at the current baud rate. The delay during
+    /// the break is just is busy-waiting.
+    pub fn send_break(&mut self, bits: u32) {
+        // Invert the TX line
+        self.register_block()
```

Comment:
I don't think this is quite right, we should be using the `tx_brk` register bit and the `txbrk_conf` (I think it's `idle_conf` on the esp32/esp32s2).

See the esp-idf impl here: https://github.com/espressif/esp-idf/blob/b5ac4fbdf9e9fb320bb0a98ee4fbaa18f8566f37/components/hal/esp32s3/include/hal/uart_ll.h#L535-L543


You can then wait for an `UART_INTR_TX_BRK_DONE` interrupt to fire to implement this in an async way, and just wait on reading the status registers for the blocking one. This completely avoids the delay issue you mentioned below.

Reply:
I didn't know about this, let me test with it a bit

Reply:
@MabezDev Ah I do know about this. Coming from IDF land, there is a function `uart_write_bytes_with_break` that will send a break **AFTER** the data is transmitted. For some use cases like mine, automotive LIN, I want to have a break at the **START** of a data transmission.

As I understand, when the `txd_brk` is set, the ESP internally doesn't act upon it until the next non-zero-length transmission finishes and then sends a break. There is no way to alter this behavior I'm aware of. 

My solution circumvents this, and works such that a break sent on TX can be detected reliably by the ESPs own break detection on the RX. (Assuming the minimum bit duration is met as mentioned above.)

Reply:
I'm going off of ESP32 technical reference manual:
https://www.espressif.com/sites/default/files/documentation/esp32_technical_reference_manual_en.pdf

- `UART_TX_DONE_INT`: Triggered when the transmitter has sent out all FIFO data.
- `UART_TX_BRK_IDLE_DONE_INT`: Triggered when the transmitter’s idle state has been kept to a minimum
after sending the last data.
- `UART_TX_BRK_DONE_INT`: Triggered when the transmitter completes sending NULL characters, after all
data in transmit-FIFO are sent.

---

These descriptions are a bit tough to understand, but I think they confirm the behavior "break gets sent after completion of non-zero-length transmission".